network/dag/state.go

```diff
@@ -93,6 +93,40 @@ func transactionCountCollector() prometheus.Counter {
 	)
 }
 
+func (s *state) initPrometheusCounters() error {
+	s.transactionCount = transactionCountCollector()
+	err := prometheus.Register(s.transactionCount)
+	if err != nil && err.Error() != (prometheus.AlreadyRegisteredError{}).Error() { // No unwrap on prometheus.AlreadyRegisteredError
+		return err
+	}
+	s.eventsNotifyCount = prometheus.NewCounter(
```

Comment:
wouldn't it make sense to split this into successfully published events and failed ones?

Reply:
Som events won't error, they just won't finish. The private payloads for example, they won't error, but will retry until finished.